Гордеев Артем

artem-gordeev opened the pull request:
Вроде все работает, осталось анимацию и тесты сделать

artem-gordeev committed:
Задание по моделированию

Вроде все работает, осталось анимацию сделать

vongostev commented:
Добрый вечер!
То, что Вы вынесли классы в отдельный файл -- это хорошо. Добавьте комментарии и проверьте PEP8.
Плюс приложите блокнот, импортирующий уже написанный код и генерирующий отчетные картинки по каждому пункту. Пока Ваш скрипт ничего не делает.

artem-gordeev committed:
Не работает анимация

Если запускать в VS Code, ложится проц после 10 картинки (комп не дает питону больше 25% использовать обычно)

artem-gordeev committed:
Не работает 

На этот раз рандомная инициализация нескольких объектов.
Однако исправлено:
- глупая ошибка v = v + a , вместо v = v + a * dt,
- разделены move_system и gravitate, теперь move_system  возращает список уничтоженных объектов, если такие есть, а gravitate сам заново запускается
- введено общее время, теперь оно не обнуляется после столкновения тел

artem-gordeev committed:
Задание по моделированию

artem-gordeev commented:
Исправлена рандомная инициализация и добавлен тест на сравнение с аналитической траекторией
Получается, остались комменты, unittest и numba

vongostev commented:
Вместо `np.array([rd.uniform(-radius,radius) for i in range (dim)])` лучше писать `rd.uniform(-radius, radius, size=dim)`

artem-gordeev committed:
Тесты скатал у Остаповича

Комментарии утром допишу, с добрым утром, кстати)

artem-gordeev commented:
Закинул туда файлы .py и .ipynb. Оба не работают, потому что VS Code не может в анимацию, Юпитер не может в доступ к кэшу, чтобы numba работала, а еще они оба не поняли, что делать в unittest, хотя я его скачал:

ERROR: C:\Users\А\AppData\Roaming\jupyter\runtime\kernel-8f9e9580-c1c3-4789-bcdd-0d886d288898 (unittest.loader._FailedTest)
----------------------------------------------------------------------
AttributeError: module '__main__' has no attribute 'C:\Users\А\AppData\Roaming\jupyter\runtime\kernel-8f9e9580-c1c3-4789-bcdd-0d886d288898'
----------------------------------------------------------------------
Ran 1 test in 0.001s
FAILED (errors=1)

Но кажется, это мелочи, связанные именно с моим компом, осталось 300 строчек комментариев написать...

artem-gordeev committed:
Задание по моделированию

Только unittest отказывается работать в Юпитере

vongostev commented:
Все неплохо, но Вы забыли про ~~бульон~~импорты, с ними было бы проще.

artem-gordeev commented:
Какие импорты вы имеете в виду?

vongostev commented:
Можно написать код в виде скрипта python, импортировать его в блокнот и просто вызвать нужные функции с анимацией. А юнит-тесты оставить в скрипте. Попробуйте. Кстати, внезапно, но если Вы раскомментируете магическую строку matplotlib, то анимация заработает.

artem-gordeev committed:
Тесты выделены в отдельный файл, импортирующий остальную программу

artem-gordeev committed:
Merge branch 'vongostev:main' into main

artem-gordeev committed:
Добавлен jupyter notebook

vongostev commented:
Артём, Вы знаете что сделайте, вынесите фигуру как отдельный аргумент. Этим Вы достигнете двух целей:
1. Если фигура равна `None`, то вообще не строится график, это для тестов без графиков.
2. Если фигура задана, то графики строятся на ней.

То есть Вы можете задать фигуру в ноутбуке и с ней работать. Так как она задана в ноутбуке, то будет нормально отображаться, и в то же время весь остальной код не надо будет сильно корежить.

artem-gordeev committed:
Подогнал тесты

Я исправил ошибки в unittest, на этом все?

artem-gordeev committed:
Merge branch 'main' of https://github.com/artem-gordeev/202-Advanced-Python-2
